module/zfs/zfs_vnops.c

```diff
@@ -1613,10 +1613,7 @@ zfs_tmpfile(struct inode *dip, vattr_t *vap, int excl,
 	dmu_tx_commit(tx);
 out:
 
-	if (error) {
-		if (zp)
-			iput(ZTOI(zp));
-	} else {
+	if (error != 0) {
```

Comment:
Let's not do this.  The coverity defect in question here is that `zp` cannot be non-NULL when `error != 0`.  This happens to be true but only due the existing implementation of the function.  That implementation could change and having the correct error handling in place here makes it less likely we'll accidentally introduce a defect in the future.  I've marking this CID as intentional in Coverity.